distribution/pull_v2.go

```diff
@@ -356,6 +360,12 @@ func (p *v2Puller) pullV2Tag(ctx context.Context, ref reference.Named) (tagUpdat
 		return false, fmt.Errorf("image manifest does not exist for tag or digest %q", tagOrDigest)
 	}
 
+	if m, ok := manifest.(*schema2.DeserializedManifest); ok {
+		if m.Manifest.Config.MediaType == schema2.MediaTypePluginConfig {
```

Comment:
It does, but it has it's own pull method. It is invalid to pull a plugin type here. My preference for the code would be rather than having to have this part of the code know about plugins, just have it check for the configuration types it expects, in this case `schema2.MediaTypeConfig`